packages/lesswrong/lib/vulcan-users/helpers.ts

```diff
@@ -70,5 +70,8 @@ export const userNumberOfItemsInPast24Hours = async function<T extends DbObject>
 };
 
 export const userFindByEmail = async function(email: string): Promise<DbUser|null> {
-  return await mongoFindOne("Users", { email: email });
+  const userByPlainEmail = await mongoFindOne("Users", { email: email });
+  if (userByPlainEmail) return userByPlainEmail;
+  const userByEmailsArray = await mongoFindOne("Users", { 'emails.address': email });
+  return userByEmailsArray;
```

Comment:
This is the function guarantees that there aren't duplicate email addresses, and is used on user signup. It's pretty important that we return users by their emails array or we'd get duplicates.